odnoklassniki-android-sdk/src/main/java/ru/ok/android/sdk/Odnoklassniki.java

```diff
@@ -124,18 +125,17 @@ private Odnoklassniki(final Context context, final String appId, final String ap
      * @param scopes      {@link OkScope} - application request permissions as per {@link OkScope}.
      * @see OkAuthType
      */
-    public final void requestAuthorization(OkListener listener, @Nullable String redirectUri,
+    public final void requestAuthorization(Activity activity, OkListener listener, @Nullable String redirectUri,
```

Comment:
The main point is to start activity from activity. mContext shouldn't keep reference for any activities.  It should use application context. That's why activity here.
And yes, listener can be removed.